crypto/x509/x509_vfy.c

```diff
@@ -2430,15 +2430,11 @@ int X509_STORE_CTX_init(X509_STORE_CTX *ctx, X509_STORE *store, X509 *x509,
         ret = X509_VERIFY_PARAM_inherit(ctx->param, store->param);
     else
         ctx->param->inh_flags |= X509_VP_FLAG_DEFAULT | X509_VP_FLAG_ONCE;
+    if (ret == 0)
+        goto err;
```

Comment:
Why do we not add an error to the stack here?

Reply:
Basically for the same reason as mentioned above: the called function, `X509_VERIFY_PARAM_inherit()` in this case, knows much better what went wrong and why.
And `ERR_R_MALLOC_FAILURE`, given here before, may be even wrong/misleading at this point.
If there is actually an malloc failure, this should be flagged by the respective allocation function itself.